### PR TITLE
Added SAML authn claim in userinfo endpoint

### PR DESCRIPTION
**Pull request by rmiccoli**

### CODE REVIEW

**Comment by rmiccoli** [comment]
```
$ curl -s -L -H "Authorization: Bearer $(ACCESS_TOKEN)" http://localhost:8080/userinfo | jq .
{
  "sub": "73f16d93-2441-4a50-88ff-85360d78c6b5",
  "name": "Admin User",
  "preferred_username": "admin",
  "given_name": "Admin",
  "family_name": "User",
  "gender": "M",
  "updated_at": 1643107340,
  "birthdate": "1950-01-01",
  "email": "1_admin@iam.test",
  "email_verified": true,
  "groups": [],
  "external_authn": [
    {
      "idpId": "https://idptestbed/idp/shibboleth",
      "attributeId": "urn:oid:1.3.6.1.4.1.5923.1.1.1.13",
      "userId": "admin@example.org"
    }
  ],
  "organisation_name": "indigo-dc"
}

```

File: iam-persistence/src/main/java/it/infn/mw/iam/persistence/model/IamUserInfo.java [hunk 1]
```diff
@@ -380,6 +391,24 @@ public JsonObject toJson() {
...
+        for(IamSamlId s : getSamlInfo()) {
+        
+          //samlinfo.add(new JsonPrimitive(s.getIdpId()));  
+          //samlinfo.add(new JsonPrimitive(s.getAttributeId())); 
+          //samlinfo.add(new JsonPrimitive(s.getUserId()));
+          sub.add("idpId", new JsonPrimitive(s.getIdpId()));
+          sub.add("attributeId", new JsonPrimitive(s.getAttributeId()));
+          sub.add("userId", new JsonPrimitive(s.getUserId()));
+        }
+        samlinfo.add(sub);
```
Comment (giacomini) [inline iam-persistence/src/main/java/it/infn/mw/iam/persistence/model/IamUserInfo.java:406]: In this way only the last one is added, isn't it?

File: iam-persistence/src/main/java/it/infn/mw/iam/persistence/model/IamUserInfo.java [hunk 2]
```diff
@@ -318,6 +323,10 @@ public void setZoneinfo(String zoneinfo) {
 
     this.zoneinfo = zoneinfo;
   }
+  
+  public void setSamlInfo(Set<IamSamlId> samlInfo) {
```
Comment (giacomini) [inline iam-persistence/src/main/java/it/infn/mw/iam/persistence/model/IamUserInfo.java:327]: Empty method?

**Comment by rmiccoli** [comment]
```
$ curl -s -L -H "Authorization: Bearer ${ACCESS_TOKEN}" http://localhost:8080/userinfo | jq .
{
  "sub": "80e5fb8d-b7c8-451a-89ba-346ae278a66f",
  "name": "Test User",
  "preferred_username": "test",
  "given_name": "Test",
  "family_name": "User",
  "gender": "M",
  "updated_at": 1643218025,
  "birthdate": "1950-01-01",
  "email": "test@iam.test",
  "email_verified": true,
  "groups": [
    "Production",
    "Analysis"
  ],
  "eduperson_principal_name": [
    "andrea.ceccanti@example.org",
    "78901@idptestbed"
  ],
  "organisation_name": "indigo-dc"
}
```

**Comment by rmiccoli** [comment]
To be reviewed..

**Comment by giacomini** [comment]
To be redone

**Pull request closed without merging**